server/src/main/scala/drt/server/feeds/api/S3ApiProvider.scala

```diff
@@ -57,21 +58,29 @@ case class S3ApiProvider(awsCredentials: AWSCredentials, bucketName: String)(imp
     )
     val zipInputStream = new ZipInputStream(inputStream)
 
-    val jsonContents = Stream
-      .continually(zipInputStream.getNextEntry)
-      .takeWhile(_ != null)
-      .map { _ =>
-        val buffer = new Array[Byte](4096)
-        val stringBuffer = new ArrayBuffer[Byte]()
-        var len: Int = zipInputStream.read(buffer)
-
-        while (len > 0) {
-          stringBuffer ++= buffer.take(len)
-          len = zipInputStream.read(buffer)
+    val jsonContents = Try {
+      Stream
+        .continually(zipInputStream.getNextEntry)
+        .takeWhile(_ != null)
+        .map { _ =>
+          val buffer = new Array[Byte](4096)
+          val stringBuffer = new ArrayBuffer[Byte]()
+          var len: Int = zipInputStream.read(buffer)
+
+          while (len > 0) {
+            stringBuffer ++= buffer.take(len)
+            len = zipInputStream.read(buffer)
+          }
+          new String(stringBuffer.toArray, UTF_8)
         }
-        new String(stringBuffer.toArray, UTF_8)
-      }
-      .toList
+        .toList
+    } match {
+      case Success(contents) => contents
+      case Failure(e) => log.error(e.getMessage)
+                         List.empty[String]
+    }
```

Comment:
won't this mean it doesn't process anything if it finds one broken file?

Reply:
or is this just for one zip file?

Reply:
This method takes a zip file. From the zip it gets the zipInputStream of each file in its contents and returns a tuple of fileName and Seq[String] - the Seq[String] is the json contents as a String for each file in the zip.

It could be possible that there is one broken file in a zip file - but then it's possible that it is an invalid zip file.

Reply:
👍 cool.